src/sentry/integrations/slack/actions/notification.py

```diff
@@ -181,57 +186,68 @@ def get_thread_ts(lifecycle: EventLifecycle) -> str | None:
-                        "action": "send_notification",
-                        "ok": e.response.get("ok", False),
-                        "status": e.response.status_code,
-                    },
-                )
-            else:
-                ts = response.get("ts")
-                new_notification_message_object.message_identifier = (
-                    str(ts) if ts is not None else None
-                )
+            # Wrap the Slack API call with lifecycle tracking
+            with MessagingInteractionEvent(
+                interaction_type=MessagingInteractionType.SEND_INCIDENT_ALERT_NOTIFICATION,
+                spec=SlackMessagingSpec(),
+            ).capture() as lifecycle:
+                try:
+                    response = client.chat_postMessage(
+                        blocks=json_blocks,
+                        text=text,
+                        channel=channel,
+                        unfurl_links=False,
+                        unfurl_media=False,
+                        thread_ts=thread_ts,
+                        reply_broadcast=reply_broadcast,
+                    )
+                    metrics.incr(
+                        SLACK_ISSUE_ALERT_SUCCESS_DATADOG_METRIC,
+                        sample_rate=1.0,
+                        tags={"action": "send_notification"},
+                    )
+                except SlackApiError as e:
+                    # Record the error code and details from the exception
+                    new_notification_message_object.error_code = e.response.status_code
+                    new_notification_message_object.error_details = {
+                        "msg": str(e),
+                        "data": e.response.data,
+                        "url": e.response.api_url,
+                    }
+
+                    log_params = {
+                        "error": str(e),
+                        "project_id": event.project_id,
+                        "event_id": event.event_id,
+                        "channel_name": self.get_option("channel"),
+                        "payload": json_blocks,
+                    }
+
+                    self.logger.info(
+                        "slack.issue_alert.error",
+                        extra=log_params,
+                    )
+                    metrics.incr(
+                        SLACK_ISSUE_ALERT_FAILURE_DATADOG_METRIC,
+                        sample_rate=1.0,
+                        tags={
+                            "action": "send_notification",
+                            "ok": e.response.get("ok", False),
+                            "status": e.response.status_code,
+                        },
+                    )
+
+                    lifecycle.add_extras(log_params)
+                    # If the error is a channel not found or archived, we can halt the flow
+                    if unpack_slack_api_error(e) in (CHANNEL_NOT_FOUND, CHANNEL_ARCHIVED):
+                        lifecycle.record_halt(e)
```

Comment:
I've been finding halts with stacktraces less helpful when searching through logs. Do you think there's value in including it here?

Reply:
yeah i don't think itll be much useful, instead going to put the reason w/o stack trace